### PR TITLE
DEV-3686: Add default page indicator to page button

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

User-facing:
- Adds a button to the pages list when a page is the default for a revenue program.
- Clicking this button shows an information popover.
- The popover can be closed with either the close button or clicking outside of it.

Code:
- Added `colors` to the `Style` type, though I left it generic for now.
- Changed the type of `default_donation_page` in `RevenueProgram` to `string | null` instead of `string | undefined`, since it's always there in API responses.
- Refactored `EditButton` into `PreviewButton`, which does basic layout of a preview + button combo, and `ContributionPageButton` and `StyleButton`.
- Revised `NewButton` to use `PreviewButton`.
- Added `ArrowPopover`, a generic component that shows a popover with an arrow pointing at some content on the page.
- Moved `assets/images/page_preview.png` into a placeholder SVG in Storybook, since it was only being used there.
- Remove `PageCard` and `StyleCard` because they weren't being used anywhere.

#### Why are we doing this? How does it help us?

Improves UX.

#### How should this be manually tested? Please include detailed step-by-step instructions.

1. Log into the admin dashboard.
2. Confirm that the page layout is correct (I changed how this works in CSS slightly).
3. Confirm pages that are not the default do not have a bookmark icon next to them.
4. Confirm pages that are the default do have a bookmark icon next to them.
5. Click the bookmark icon next to a page.
6. Confirm the popover that appears matches Figma.
7. Confirm the link in the popover is correct.
8. Click the close button in the popover and confirm it closes the popover.
9. Repeat steps 1-8, but click somewhere outside the popover and confirm that that closes the popover too.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

Yes, we need to document the bookmark icon.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

Updated docs.

#### Are there any smells or added technical debt to note?

The Figma designs show a rounded corner on the popover arrow, but I couldn't figure out a way to implement that.

There's an odd Jest warning in the ArrowPopover tests I wasn't able to figure out.

```
FakeTimers: cancelAnimationFrame was invoked to clear a native timer instead of one created by this library.
To automatically clean-up native timers, use `shouldClearNativeTimers`.
```

I tried a bunch of techniques, like starting fake timers at different points in the process, but couldn't resolve this.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-3686

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

Rest of stories in https://news-revenue-hub.atlassian.net/browse/DEV-3056.

### CODE REVIEW

File: spa/src/components/content/pages/AddPage.tsx [hunk 1]
```diff
@@ -106,7 +106,7 @@ function AddPage() {
 
   return (
     <>
-      <NewButton buttonTestId="new-page-button" onClick={handleClick} />
+      <NewButton ariaLabel="New Page" data-testid="new-page-button" label="New Page" onClick={handleClick} />
```
Comment (GuiMend) [inline spa/src/components/content/pages/AddPage.tsx:109]: Don't think this `ariaLabel` is doing anything. The correct attribute name is `aria-label`

Comment (nrh-cklimas) [reply spa/src/components/content/pages/AddPage.tsx:109]: This is a prop on PreviewButton. We could rename that prop to `aria-label` if you think that would make things clearer.

File: spa/src/components/common/Button/ContributionPageButton/ContributionPageButton.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+export interface ContributionPageButtonProps
+  extends Omit<PreviewButtonProps, 'corner' | 'disabled' | 'label' | 'preview'> {
+  page: ContributionPage;
+}
+
+export function ContributionPageButton({ page, ...other }: ContributionPageButtonProps) {
+  const label =
+    page.id === page.revenue_program.default_donation_page ? (
+      <Label>
+        <DefaultPageButton domId={page.name.replace(/\s/g, '-') + '-default-page'} /> {page.name}
```
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/common/Button/ContributionPageButton/ContributionPageButton.tsx:23]: It's a prop on DefaultPageButton -- could make it just `id`.

File: spa/src/components/content/pages/AddPage.tsx (hunk 1 above)
Comment (GuiMend) [reply spa/src/components/content/pages/AddPage.tsx:109]: Oh ok. Didn't see those 2 prop names. We can leave it as is 👍

File: spa/src/components/common/Button/ContributionPageButton/ContributionPageButton.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/common/Button/ContributionPageButton/ContributionPageButton.tsx:23]: It's ok 👍

**Pull request merged**